### PR TITLE
Removed declarative CAS permission attributes and comments (Part 2)

### PR DESCRIPTION
**Pull request by ojhad**
Part of issue #241

### CODE REVIEW

**Comment by vatsan-madhavan** [comment]
lgtm. 

did you get a chance to run an IL-diff on the resultant assemblies and the pre-change assemblies?

**Comment by stevenbrix** [comment]
looks good!

**Comment by ojhad** [comment]
> lgtm.
> 
> did you get a chance to run an IL-diff on the resultant assemblies and the pre-change assemblies?

I didn't get a chance to do that yet, but I will run it before merging this PR

File: src/Microsoft.DotNet.Wpf/src/System.Xaml/GlobalSuppressions.cs [hunk 1]
```diff
@@ -124,8 +124,6 @@
 #endregion
 
 #region Microsoft.Security Suppressions
-[module: SuppressMessage("Microsoft.Security", "CA2103:ReviewImperativeSecurity", Scope = "member", Target = "System.Xaml.Permissions.XamlLoadPermission.#Copy()", Justification = "Reviewed by senior CLR security developer.")]
 [module: SuppressMessage("Microsoft.Security", "CA2103:ReviewImperativeSecurity", Scope = "member", Target = "MS.Internal.Xaml.Runtime.DynamicMethodRuntime.#.ctor(MS.Internal.Xaml.Runtime.XamlRuntimeSettings,System.Xaml.XamlSchemaContext,System.Xaml.Permissions.XamlAccessLevel)", Justification = "Reviewed by Microsoft.")]
```
Comment (rladuca) [inline src/Microsoft.DotNet.Wpf/src/System.Xaml/GlobalSuppressions.cs]: Why the others and not the 2 leftover suppressions?

File: src/Microsoft.DotNet.Wpf/src/DirectWriteForwarder/CPP/DWriteWrapper/TextAnalyzer.cpp [hunk 2]
```diff
@@ -297,7 +277,6 @@ namespace MS { namespace Internal { namespace Text { namespace TextInterface
     }
 
 // Warning 4714 (__forceinline function not inlined)
-// is expected here because TextAnalyzer::GetGlyphs is marked with [SecurityCritical]
```
Comment (rladuca) [inline src/Microsoft.DotNet.Wpf/src/DirectWriteForwarder/CPP/DWriteWrapper/TextAnalyzer.cpp]: This comment is strange now without this line.

File: src/Microsoft.DotNet.Wpf/src/DirectWriteForwarder/CPP/DWriteWrapper/TextAnalyzer.cpp [hunk 3]
```diff
@@ -276,19 +276,6 @@ namespace MS { namespace Internal { namespace Text { namespace TextInterface
         }
     }
 
-// Warning 4714 (__forceinline function not inlined)
```
Comment (rladuca) [inline src/Microsoft.DotNet.Wpf/src/DirectWriteForwarder/CPP/DWriteWrapper/TextAnalyzer.cpp]: I'm not so sure about suddenly inlining a set of function that were not inlined previously.  I understand the original reasoning for them not being inlined has gone away, but we need to be 100% certain there are no adverse affects from doing this.

Can we hold off on this, file an issue and do a more thorough examination of these changes with a larger test pass later?

Comment (rladuca) [reply src/Microsoft.DotNet.Wpf/src/DirectWriteForwarder/CPP/DWriteWrapper/TextAnalyzer.cpp]: I think I misunderstood this, it's just disabling a warning, not enabling inlining, resolving.

**Comment by vatsan-madhavan** [comment]
I did a quick IL-diff scan and the changes look ok to me. There is some compiler generated code in DirectWriteForwarder that uses `SecurityCriticalAttribute`  (for e.g., global method `_atexit_m`). We need to figure out a way to resolve these dependencies.

**Comment by ojhad** [comment]
> I did a quick IL-diff scan and the changes look ok to me. There is some compiler generated code in DirectWriteForwarder that uses `SecurityCriticalAttribute` (for e.g., global method `_atexit_m`). We need to figure out a way to resolve these dependencies.

Will take a look and include this in the last PR

**Pull request merged**